### PR TITLE
environmentd: remove unused TLS modes

### PR DESCRIPTION
**Pull request by mjibson**
We've committed to the cloud and will not be using anything besides that and local dev.

### Motivation

  * This PR adds a feature that has not yet been specified.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: src/environmentd/src/http.rs [hunk 1]
```diff
@@ -78,7 +76,6 @@ pub struct TlsConfig {
 #[derive(Debug, Clone, Copy)]
 pub enum TlsMode {
     Require,
-    AssumeUser,
 }
```
Comment (benesch) [inline src/environmentd/src/http.rs:79]: Seems like perhaps this should just become a bool?

File: src/pgwire/src/server.rs [hunk 2]
```diff
@@ -64,9 +64,6 @@ pub struct TlsConfig {
 pub enum TlsMode {
     /// Clients must negotiate TLS encryption.
     Require,
```
Comment (benesch) [inline src/pgwire/src/server.rs:66]: Ditto.

**Review by benesch — APPROVED** [review]

File: src/environmentd/src/http.rs (hunk 1 above)
Comment (mjibson) [reply src/environmentd/src/http.rs:79]: Are you opposed to a two variant enum?

Comment (benesch) [reply src/environmentd/src/http.rs:79]: `TlsMode::Disabled` and `TlsMode::Enabled`? That seems preferable to a one variant enum, for sure!

Comment (mjibson) [reply src/environmentd/src/http.rs:79]: We can probably get rid of each service having its own TlsMode also, unless you know of a reason to keep that.

Comment (benesch) [reply src/environmentd/src/http.rs:79]: There's no good place to put a shared definition, is why it is the way that it is.

Comment (mjibson) [reply src/environmentd/src/http.rs:80]: Done.

**Review by benesch — APPROVED** [review]

**Pull request merged**